pkg/application/app.go

```diff
@@ -620,7 +623,8 @@ func (app *Avalanche) GetBlockchainNamesOnNetwork(network models.Network) ([]str
 				}
 			}
 		}
-		if sc.Networks[networkName].BlockchainID != ids.Empty {
+		sovKindCriteria := !onlySOV || onlySOV && sc.Sovereign
```

Comment:
`!onlySOV || onlySOV` always returns `true`

Reply:
did you mean `!onlySOV || sc.Sovereign` ?

Reply:
&& has precedence over ||, this is the same as "!onlySOV || (onlySOV && sc.Sovereign)"